### ISSUE TITLE
No formatter installed

### ISSUE THREAD

**Issue by Sytten** [hashicorp/vscode-terraform#416]
Before the code was formatting properly, but the new extension did not register the formatter?

**Comment by audiolomb** [comment]
Same thing here. Format on save does not work anymore in 2.0.2.

**Comment by sephross** [comment]
I would also request that the vscode extension introduces a formatter that is not dependant on the language server. This will enable users who do not have, or have not initialized terraform on their local machine a means of formatting tf files without running into language server related problems.

**Comment by paulmilbank** [comment]
This seems to work again with the new version of the language server installed as per https://github.com/hashicorp/vscode-terraform/issues/417#issuecomment-650228463

**Comment by Skaronator** [comment]
Just upgraded the terraform language server to 0.5.1 and it works again! ;)

**Comment by aicarmic** [comment]
Thanks for the report, as suggested this issue was addressed in 0.5.1.

**Comment by radeksimko** [comment]
Just to clarify auto-formatting may need to be configured in VS Code and it is configured via VS Code settings, not extension settings anymore: https://code.visualstudio.com/docs/getstarted/settings#_language-specific-editor-settings
e.g.

```json
    "[terraform]": {
        "editor.formatOnSave": true
    }
```
The extension will then auto-format on save via the language server.

The language server gets installed automatically and allows us to deal with version and other language differences and complexities in a single place. This is a sustainable model where we can avoid maintaining duplicate logic between VS Code, many other IDE extensions _and_ the language server. As a result the language server will be the only supported way to do formatting in the foreseeable future. If you run into issues with the language server [please do report them](https://github.com/hashicorp/terraform-ls/issues/new/choose), so we can address them.
